src/operator/controllers/monitor.go

```diff
@@ -325,6 +325,31 @@ func getNATSState(client HTTPClient, pods *concurrentPodMap) *vizierState {
 	return okState()
 }
 
+// getEtcdState determines the state of etcd then translates that
+// to a corresponding VizierState.
+func getEtcdState(pods *concurrentPodMap) *vizierState {
+	pods.mapMu.Lock()
+	defer pods.mapMu.Unlock()
+
+	unlabeledPods, ok := pods.unsafeMap[""]
+	if !ok {
+		return &vizierState{Reason: status.EtcdPodsMissing}
+	}
+
+	for podName, pod := range unlabeledPods {
+		if !strings.HasPrefix(podName, "pl-etcd-") {
+			continue
+		}
+		for _, c := range pod.pod.Status.ContainerStatuses {
+			if c.State.Waiting != nil && c.State.Waiting.Reason == "CrashLoopBackOff" {
+				return &vizierState{Reason: status.EtcdPodsCrashing}
+			}
+		}
```

Comment:
should we check for a certain number of restarts and number of pods that are in this state?